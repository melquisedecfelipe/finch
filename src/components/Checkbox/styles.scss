.item-check {
  display: flex;
  justify-content: center;
  align-items: center;

  > label {
    position: relative;
    display: inline-block;
    width: 26px;
    height: 15px;

    > input {
      opacity: 0;
      width: 0;
      height: 0;
    }

    > span {
      position: absolute;
      cursor: pointer;
      top: 0;
      left: 0;
      right: 0;
      bottom: 0;
      background-color: #ccc;
      -webkit-transition: 0.4s;
      transition: 0.4s;
      border-radius: 34px;

      &:before {
        position: absolute;
        content: '';
        height: 10px;
        width: 10px;
        left: 3px;
        bottom: 3px;
        background-color: white;
        -webkit-transition: 0.4s;
        transition: 0.4s;
        border-radius: 50%;
      }
    }

    input:checked + span {
      background-color: rgb(0, 134, 248);
    }

    input:checked + span:before {
      -webkit-transform: translateX(10px);
      -ms-transform: translateX(10px);
      transform: translateX(10px);
    }
  }

  & p {
    font-size: 12px;
    line-height: 21px;
    color: #8e879e;
    margin-left: 5px;
  }
}
