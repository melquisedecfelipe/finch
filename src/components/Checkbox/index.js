import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import './styles.scss';

import { Creators as ItemActions } from '../../store/ducks/item';

function Checkbox({ id, favorito, setFavorite }) {
  function handleCheckbox() {
    setFavorite(id);
  }

  return (
    <div className="item-check">
      <label htmlFor={id}>
        <input
          type="checkbox"
          id={id}
          value={favorito}
          onChange={() => handleCheckbox()}
          defaultChecked={favorito === true}
        />
        <span />
      </label>
      <p>tornar favorito</p>
    </div>
  );
}
const mapDispatchToProps = dispatch => bindActionCreators(ItemActions, dispatch);

export default connect(
  null,
  mapDispatchToProps,
)(Checkbox);
