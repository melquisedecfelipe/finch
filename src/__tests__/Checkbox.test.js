import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Checkbox from '../components/Checkbox';

import state from '../services/mockStore';

describe('Checkbox testing with Enzyme', () => {
  const mockStore = configureStore();
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(state);
    wrapper = shallow(<Checkbox store={store} />);
  });

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
