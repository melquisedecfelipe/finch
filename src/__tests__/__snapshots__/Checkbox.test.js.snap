// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox testing with Enzyme renders without crashing 1`] = `ShallowWrapper {}`;
